### PR TITLE
Starting the client-side JS fetching of project data

### PR DESCRIPTION
**Pull request by chums923**
Hi all

So I was a part of some brief discussion about including project data in the search results page. There was some good talk, but I thought things would be a little easier if I just got an MVP out there and we could talk about it. I decided to go with a purely client-side solution because it would require the least amount of tempering with the back-end, but that's definitely a route we can go down if people think it's better.

To check this out, search for anything basically. Then find a bug that is being tracked using the Github Issue Tracker. Expand that bug, then look for the "See project data" button right under the "More about this project" link. Click it to check out a modal that pops up with data gleaned from the Github API. It's pretty basic right now. But I wanted to get some feedback before I:
- Added more and cooler data from Github
- Added other issue trackers besides Github
- Clean up my code (but it's not unreadable at this point :) )
- Write tests (pretty trivial anyway)
- Optimize for API request limits
- Squash my commits

Here's a screenshot of one example

image

Feel free to email me (chumbleya@gmail.com) if you have any questions/comments. Thanks!

### CODE REVIEW

**Comment by paulproteus** [comment]
This is pretty cool. I will try to review in more depth in the next few days; if someone else beats me to it, that'd be fine too. (-:

File: mysite/static/js/search/apiWrapper.js [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+	this.executeAPI = function() {
+		var URL = this.endpoint+this.user+"/"+this.repo;
+		$.get(URL, function(data){
+			_this.populateModal(data);
+		});
+	};
+
+	this.populateModal = function(data) {
+		console.log(data);
```
Comment (albertyw) [inline mysite/static/js/search/apiWrapper.js]: extra `console.log`?

**Comment by chums923** [comment]
@albertyw Yes, definitely. Thanks for catching that, I should have taken that out. But, this PR isn't something I'm looking to get merged into the live site. I'm just looking to get some feedback on this feature to see if people want me to finish it :)

**Comment by ehashman** [comment]
I'm not able to review this right now, but there are a bunch of spurious merge commits in this PR. I recommend you use `git rebase` to remove them before a final review.

**Comment by paulproteus** [comment]
Hi @chums923 ,

Sorry about the _long_ delay in getting back to you about this. Let me look into this now and see what feedback I can give.

If you're interested in reviving this branch, I'd be very happy to hear that. Let me know if you are, or aren't, so I know what to expect, and sorry again about the latency.

File: mysite/static/js/search/apiWrapper.js [hunk 2]
```diff
@@ -0,0 +1,643 @@
...
+	TODO: 
+		1) Finish Python BugTracker
+		2) Make timeout on spinny wheel thing
+			Does that mean timeout on request too?
+			Fill the modal with a different message
+*/
+
+
+/**
+DataCache is a local (not currently using cookies) record of which APIs
```
Comment (paulproteus) [inline mysite/static/js/search/apiWrapper.js]: In the long run, if you want to store data locally, consider using `localStorage` rather than cookies, for what it's worth.

http://people.rit.edu/tlh4225/736/exercises/storageWriteup.html explains the differences.

**Comment by paulproteus** [comment]
This pull request is a bunch of really cool ideas. Some quick thoughts:

* The more data we fetch via these API calls to existing services where projects already publish data, the better. I think fairly few projects write information about themselves on the OpenHatch site, so it's unwise for us to rely on that.

* This code makes it pretty clear how to add support for a variety of bug trackers, which is cool.

Some concerns & proposals I have:

* The current bug crawling backend, whatever its flaws, does have reasonably good automated test coverage, and I believe that test coverage has done a reasonably good job of maintaining code quality over time. Therefore, I'd love to see this code gain automated tests via e.g. http://blog.toast38coza.me/getting-started-with-jasmine-tests-and-django/

* The proposed interface creates a pop-up. I propose instead that we always show this information in the sidebar, right above the words "More about this project". Maybe show a spinner while waiting for the API fetch. That way, it requires less user interaction.

@chums923 what do you think of the above? Sorry about the long delay, and sorry for misunderstanding your proposal long ago.

**Comment by paulproteus** [comment]
For now I'm going to mark this `needs-revision` so that I can easily remember that I've already reviewed it once.

**Comment by chums923** [comment]
Hi @paulproteus, thanks for getting back to this branch. 

So, I haven't really had any time to work or think about this in a while. I did all of this as a case study in a class I was taking this past semester. I would be interested in getting back to this when I get back to school next month, depending on what we're talking about for time commitment. 

I 100% agree that this needs automated testing. I also agree that popups are pretty lame, I just didn't want to mess too much with the HTML when I was doing this - I was trying to make this as self-contained as possible. 

A fear I have in general is that there are so many projects using proprietary source control or otherwise obscure ones. That fear is mitigated a bit by the fact that it projects are moving to Github every day, it seems like.  

So, in short, I'm game to continue thinking and talking about this. But, I can't really be relied upon too heavily in terms of time commitment. Thanks again!
